tgui/packages/tgui/interfaces/JobSelection.tsx

```diff
@@ -1,31 +1,30 @@
 import { useBackend } from '../backend';
-import { Button, DepartmentEntry, DepartmentPane, Icon, Stack } from '../components';
+import { Button, DepartmentEntry, DepartmentPane, Icon } from '../components';
 import { Window } from '../layouts';
 import { Color } from 'common/color';
-import { merge } from 'common/merge';
 import { SFC } from 'inferno';
 import { JobToIcon } from './common/JobToIcon';
-import { BaseDepartmentInfo, BaseJobInfo } from '../components/DepartmentPane';
+import { BaseDepartmentInfo } from '../components/DepartmentPane';
+import { deepMerge } from 'common/collections';
 
-class Job implements BaseJobInfo {
+type Job = {
   unavailable_reason: string | null;
   command: boolean;
   open_slots: number;
   used_slots: number;
   icon: string;
   prioritized: boolean;
   description: string;
-}
+};
 
-class Department implements BaseDepartmentInfo {
-  color: string;
+type Department = {
   open_slots: number;
-  jobs: { [x: string]: Job };
-}
+  jobs: Record<string, Job>;
+} & BaseDepartmentInfo;
```

Comment:
Order of `&` actually matters, `BaseDepartmentInfo` most likely overwrites `jobs` with `Record<string, any>`.

Reply:
Other way around, it seems to avoid overwriting existing names.

Reply:
Try this instead:

```tsx
type BaseDepartment<Job = any> = {
  jobs: Record<string, Job>;
};

type Job = { x: string };

type Department = BaseDepartment<Job> & {
  open_slots: number;
};
```